elements/rh-secondary-nav/rh-secondary-nav.ts

```diff
@@ -0,0 +1,389 @@
+ * @slot logo           - Logo added to the main nav bar, expects a `<a> | <a><svg/></a> | <a><img/></a>`
+ * @slot nav            - Navigation list added to the main nav bar, expects a `<ul>`
+ * @slot cta            - Nav bar level CTA, expects a `<pfe-cta>
+ *
+ * @csspart base        - {HTMLNavElement} container, <nav> element
+ * @csspart container   - {HTMLElement} css grid container, <div> element
+ * @csspart cta         - {HTMLElement} container, <div> element
+ *
+ * @fires { SecondaryNavOverlayEvent }  - Fires when an dropdown is opened or closed in desktop view or when
+ *                                        the mobile menu button is toggled in mobile view.
+ */
+@customElement('rh-secondary-nav') @pfelement()
+export class RhSecondaryNav extends LitElement {
+  static readonly version = '{{version}}';
+
+  static readonly styles = [styles];
+
+  #logger = new Logger(this);
+
+  /**
+   * executes this.shadowRoot.querySelector('rh-secondary-nav-overlay')
+   */
+  @query('rh-secondary-nav-overlay') _overlay: RhSecondaryNavOverlay | undefined;
+
+  /**
+   * executes this.shadowRoot.querySelector('#container')
+   */
+  @query('#container') _container: HTMLElement | undefined;
+
+  /**
+   * executes this.shadowRoot. querySelctor('nav');
+   */
+  @query('nav') _nav: HTMLElement | undefined;
+
+  /**
+   * executes this.shadowRoot.querySelector('button')
+   */
+  @query('button') _mobileMenuButton: HTMLButtonElement | undefined;
+
+
+  /**
+   * `isMobile` property is true when viewport `(min-width: ${tabletLandscapeBreakpoint})`.
+   * Property is observed for changes, and its value is updated using matchMediaControler
+   * when viewport changes at breakpoint or first load of the component.
+   */
+  @observed
+  @property({ reflect: false }) _isMobile = false;
+
+  /**
+   * Checks if passed in element is a RhSecondaryNavDropdown
+   * @param element:
+   * @returns {boolean}
+   */
+  static isDropdown(element: Element|null): element is RhSecondaryNavDropdown {
+    return element instanceof RhSecondaryNavDropdown;
+  }
+
+  /**
+   * Updates isMobile property with matchMedia when viewport changes
+   */
+  protected matchMedia = new MatchMediaController(this, `(min-width: ${tabletLandscapeBreakpoint})`, {
+    onChange: ({ matches }) => {
+      this._isMobile = !matches;
+    }
+  });
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('change', this._dropdownChangeHandler as EventListener);
+    this.addEventListener('overlay-change', this._toggleNavOverlay as EventListener);
+    this.addEventListener('focusout', this._focusOutHandler);
+    this.removeAttribute('role');
+  }
+
+  firstUpdated() {
+    // after update the overlay should be available to attach an event listener to
+    this._overlay?.addEventListener('click', this._overlayClickHandler as EventListener);
+    this._isMobile = (window.outerWidth < parseInt(tabletLandscapeBreakpoint.toString().split('px')[0]));
```

Comment:
if this is needed, i'd rather see it calculated in `updated`, or in a (new) callback in RHDSScreenSizeController

Reply:
My issue here isn't that the initial implementation of `RHDSScreenSizeController` is sacrosanct or anything - by all means improve it! My only issue here is doing this logic inside the component. If you need to make changes to `RHDSScreenSizeController` to get this feature in secondary-nav, go for it, I just don't want to see breakpoints imported or hard coded into components.

The reason why this is important is so that we maintain some level of uniformity and control over the design system. This is also the reason we created the tokens repo. We could very easily loosen this discipline, but that would lead to a situation where each individual component[ author] makes their own design assumptions, and keeps them private and local to line N in whichever component file. Even in the short term, this degrades maintainability.

So to answer your question in the google doc, yes, we should eventually put those components in the tokens repo and import them into `RHDSScreenSizeController.ts`. If, for the time being, you need to hard code values there or import them from `lib/tokens.ts`, no problemo, it's still less to worry about than having those token references scattered about in component lifecycle methods.

> But secondary-nav uses values not represented in tokens repo or tokens.ts

Add those values there. Components should not own design tokens.

> But I need to get this out for secondary nav now

Add the values in `RHDSScreenSizeController.ts`, in this PR. Make sure it doesn't break table. This shouldn't be a blocker, from what you've described, the problem is the internal implementation of `RHDSScreenSizeController`, which is very much open to change and improvement.

My _only_ requirements here are that breakpoints be handled exclusively in the controller, and that the component APIs be declarative (i.e., that components aren't doing any comparisons against token values). This is so that tokens are
1. available to any component while being owned by none
2. defined (if even initially aspirationally/locally) in the tokens
3. themeable

Reply:
Fixed in 7996deb4adcfb81f26ebf40d05427e5bb55abb5c and c03d4820dbd8a6b95ee2520b3a7f5c4968b6e766